features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/ElasticFlowRepository.java

```diff
@@ -167,7 +170,7 @@
      *
      * This maps a node ID to a set if snmpInterface IDs.
      */
-    private final ConcurrentMap<Integer, Set<Integer>> markerCache = Maps.newConcurrentMap();
+    private Map<Direction, Cache<Integer, Set<Integer>>> markerCache = new TreeMap<>();
```

Comment:
(Beside the missing `final` ...) a `TreeMap` used for two variables seems to be overkill. There are cases of conditional handling below anyways.
A `Map#get()` and `Map#put()` compared to reading and writing two variable seems to be wrong.
If we decide to keep the `Map`, there is `EnumMap`...